pkg/publicshare/manager/json/json.go

```diff
@@ -343,10 +343,22 @@ func (m *manager) UpdatePublicShare(ctx context.Context, u *user.User, req *link
 		old, _ := json.Marshal(share.Permissions)
 		new, _ := json.Marshal(req.Update.GetGrant().Permissions)
 
-		if m.writeableShareMustHavePassword && publicshare.IsWriteable(req.GetUpdate().GetGrant().GetPermissions()) && !share.PasswordProtected {
+		if m.writeableShareMustHavePassword &&
+			publicshare.IsWriteable(req.GetUpdate().GetGrant().GetPermissions()) &&
+			(!share.PasswordProtected && req.GetUpdate().GetGrant().GetPassword() == "") {
```

Comment:
I don't understand why we check for `!share.PasswordProtected`. Why is it relevant? 

```suggestion
		if m.writeableShareMustHavePassword &&
			publicshare.IsWriteable(req.GetUpdate().GetGrant().GetPermissions()) &&
			req.GetUpdate().GetGrant().GetPassword() == "" {
```

Reply:
this is relevant when the password has already been set before. Then we are able to change the permissions without a new password.

Reply:
You only need to provide a new password when we have no existing password.